### ISSUE TITLE
HeaderBackButton components triggers accessibility issue while default does not

### ISSUE THREAD

**Issue by VariabileAleatoria** [react-navigation/react-navigation#12509]
### Current behavior

When using React Navigation’s default back button, Google Accessibility Scanner does not report any issues.  However, when manually setting headerLeft using HeaderBackButton, the scanner reports:
> "Consider making this clickable item larger. This item's size is 30dp x 30dp. Consider making this touch target 48dp wide and 48dp high or larger."

![Image](https://github.com/user-attachments/assets/ea0c45d6-05e2-4abc-8d5b-9e6d32d838b1)
![Image](https://github.com/user-attachments/assets/2f0c25bc-97e8-4bad-af67-587767fb7fdf)

### Expected behavior

 Both implementations should have the same touch target size and pass accessibility checks.

### Reproduction

https://snack.expo.dev/k_IhBrcLAKDGt5mNzBgeA

### Platform

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [x] @react-navigation/native-stack
- [ ] react-native-drawer-layout
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native                      |   7.0.18
| @react-navigation/native-stack           | .     *
| react-native-screens                            | ~4.4.0
| react-native-safe-area-context           | 4.12.0
